src/database/statistics/dateUtil.ts

```diff
@@ -85,62 +56,158 @@ export function spreadsGroupedByTimeslots(timeRepoPairs: {
+      weekSpreadSum += dates.weekSpread[currentWeek].length;
+    } else {
+      // add repo Ids to current week entry and make set for no duplicates
+      const currentRepos = dates.weekSpread[currentWeek];
+      const mergedRepos = [].concat(currentRepos, repoArr);
+      dates.weekSpread[currentWeek] = Array.from(new Set(mergedRepos));
     }
-    if (addDays(sprintDate, 14) <= new Date(timestamps[i])) {
-      dates.sprintSpread[sprintDate] = Array.from(sprintSpread);
-      sprintSpreadSum += sprintSpread.size;
-      sprintSpread.clear();
-      sprintDate = timestamps[i];
-    }
-    if (addDays(monthDate, 30) <= new Date(timestamps[i])) {
-      dates.monthSpread[monthDate] = Array.from(monthSpread);
-      monthSpreadSum += monthSpread.size;
-      monthSpread.clear();
+    if (!monthsAreEqual(currentMonth, nextMonth)) {
+      dates.monthSpread[nextMonth] = repoArr;
       monthDate = timestamps[i];
+      monthSpreadSum += dates.monthSpread[currentMonth].length;
+    } else {
+      // add repo Ids to current month entry and make set for no duplicates
+      const currentRepos = dates.monthSpread[currentMonth];
+      const mergedRepos = [].concat(currentRepos, repoArr);
+      dates.monthSpread[currentMonth] = Array.from(new Set(mergedRepos));
+    }
+    // add the current repo amounts for next week/month in last step
+    // as they are not considered otherwise
+    if (i == timestamps.length - 1) {
+      weekSpreadSum += dates.weekSpread[nextWeek].length;
+      monthSpreadSum += dates.monthSpread[nextMonth].length;
     }
   }
-  // after all, add the computed numbers to the dates object
+
   dates.daySpreadSum = daySpreadSum;
   dates.weekSpreadSum = weekSpreadSum;
-  dates.sprintSpreadSum = sprintSpreadSum;
   dates.monthSpreadSum = monthSpreadSum;
   dates.days = Object.keys(dates.daySpread).length;
   dates.weeks = Object.keys(dates.weekSpread).length;
-  dates.sprints = Object.keys(dates.sprintSpread).length;
   dates.months = Object.keys(dates.monthSpread).length;
   return dates;
 }
+
+/**
+ * Put in a date @param strDate and get the
+ * calender week number as @return.
+ * @source https://weeknumber.com/how-to/javascript
+ */
+function getCalendarWeek(strDate: string): number {
+  const date = new Date(strDate);
+  date.setHours(0, 0, 0, 0);
+  // Thursday in current week decides the year.
+  date.setDate(date.getDate() + 3 - ((date.getDay() + 6) % 7));
+  // January 4 is always in week 1.
+  var week1 = new Date(date.getFullYear(), 0, 4);
+  // Adjust to Thursday in week 1 and count number of weeks from date to week1.
+  return (
+    1 +
+    Math.round(
+      ((date.getTime() - week1.getTime()) / 86400000 -
+        3 +
+        ((week1.getDay() + 6) % 7)) /
+        7,
+    )
+  );
+}
+
+/**
+ * If two weeks @param week1 and @param week2 are equal,
+ * @return true, else false.
+ */
+function weeksAreEqual(week1: number, week2: number): boolean {
+  if (week1 == week2) {
+    return true;
+  }
+  return false;
+}
```

Comment:
Always nice to give conditions a name, but you can just reduce this to
```ts
return week1 == week2
```

Reply:
oh yes. Done!